apps/prairielearn/elements/pl-number-input/pl-number-input.py

```diff
@@ -217,55 +240,39 @@ def render(element_html: str, data: pl.QuestionData) -> str:
         info_params["allow_complex"] = pl.get_boolean_attrib(
             element, "allow-complex", ALLOW_COMPLEX_DEFAULT
         )
-        info_params["show_info"] = pl.get_boolean_attrib(
-            element, "show-help-text", SHOW_HELP_TEXT_DEFAULT
-        )
+        info_params["show_info"] = show_help_text
         info_params["allow_fractions"] = allow_fractions
 
         # Find the true answer to be able to display it in the info popup
-        ans_true = None
-        if pl.get_boolean_attrib(
-            element, "show-correct-answer", SHOW_CORRECT_ANSWER_DEFAULT
-        ):
-            ans_true = format_true_ans(element, data, name)
-        if ans_true is not None:
-            info_params["a_tru"] = ans_true
-
-        with open(NUMBER_INPUT_MUSTACHE_TEMPLATE_NAME, "r", encoding="utf-8") as f:
-            info = chevron.render(f, info_params).strip()
-        with open(NUMBER_INPUT_MUSTACHE_TEMPLATE_NAME, "r", encoding="utf-8") as f:
-            info_params.pop("format", None)
-            # Within mustache, the shortformat generates the placeholder that is used as a placeholder inside of the numeric entry.
-            # Here we opt to not generate the value, hence the placeholder is empty.
-            # The placeholder text may be overriden by setting the 'placeholder' attribute in the pl-number-input HTML tag
-            if pl.has_attrib(element, "placeholder"):
-                # 'placeholder' attribute is set, override the placeholder text
-                html_params["placeholder"] = pl.get_string_attrib(
-                    element, "placeholder"
-                )
-            else:
-                # 'placeholder' attribute not set, use default shortformat as placeholder text
-                info_params["shortformat"] = pl.get_boolean_attrib(
-                    element, "show-placeholder", SHOW_PLACEHOLDER_DEFAULT
-                )
-                html_params["placeholder"] = chevron.render(f, info_params).strip()
+        if show_correct_answer:
+            html_params["a_tru"] = format_true_ans(element, data, name)
+
+        info = chevron.render(template, info_params).strip()
+
+        info_params.pop("format", None)
```

Comment:
Why do we do this?

Reply:
It's because this parameters dict gets reused between rendering the info panel and the placeholder attribute in the fallback case (which is mimicking previous behavior). It's pretty ugly but we left it this way to minimize the number of changes from the original file. I think this can be cleaned up slightly however, I'll try to simplify it a little bit when I address the other items.

Reply:
Sounds good!

Reply:
Simplified a few things here: https://github.com/PrairieLearn/PrairieLearn/pull/8435/commits/682198bbf6fc86d79dacbfee1c8418b6ba91cd64